src/picosvg/svg.py

```diff
@@ -944,6 +948,69 @@ def _apply_gradient_translation(self, inplace=False):
             else:
                 del el.attrib["gradientTransform"]
 
+    def _resolve_gradient_templates(self, inplace=False):
+        # Gradients can have an 'href' attribute that specifies another gradient as
+        # a template, inheriting its attributes and/or stops when not already defined:
+        # https://www.w3.org/TR/SVG/pservers.html#PaintServerTemplates
+        if not inplace:
+            svg = SVG(copy.deepcopy(self.svg_root))
+            svg._resolve_gradient_templates(inplace=True)
+            return svg
+
+        href_attr = _xlink_href_attr_name()
+        el_by_id = {el.attrib["id"]: el for el in self.xpath(".//svg:*[@id]")}
+
+        def resolve_gradient_template(gradient):
```

Comment:
the nested relatively long function isn't helping with readability much, pop it out?

Reply:
typing gradient would be nice too

Reply:
The nested function is to be able to recurse